fs-v1-multipart.go

```diff
@@ -321,23 +321,26 @@ func (fs fsObjects) PutObjectPart(bucket, object, uploadID string, partID int, s
 	}
 	var buf = make([]byte, int(bufSize))
 
-	// Read up to required size
-	totalLeft := size
+	bytesWritten := int64(0)
+	// Read the buffer till io.EOF and append the read data to the temporary file.
 	for {
-		n, err := io.ReadFull(limitDataReader, buf)
-		if err == io.EOF {
-			break
-		}
-		if err != nil && err != io.ErrUnexpectedEOF {
-			return "", toObjectErr(err, bucket, object)
+		n, rErr := limitDataReader.Read(buf)
```

Comment:
.Read is not efficient for us, it will lead to many more calls since Read also doesn't return full 'buf' bytes.  io.ReadFull was used for this purpose only. 

Also move this into separate function as 'fs-createfile.go'